constellations/gemini-admin/src/classes/classEdit.js

```diff
@@ -206,8 +212,49 @@ export class ClassEditPage extends React.Component {
         });
     };
 
-    render = () => {
-        const title = this.state.isEdit ? "Edit Class" : "Add Class";
+    renderModal = (
+        showSaveModal,
+        showDeleteModal,
+        onModalOkSaved,
+        onModalDeleteConfirm,
+        onModalDismiss
+    ) => {
```

Comment:
I agree handling modals per page can be a bit messy, especially because every page can potentially have many diff modals. I wonder if we should remove all class-related modals to a diff file? Or even since modals across all pages are similar, wonder if we can move it to the utils folder?? (just lightly thinking about it).

For this case though, I think it would make more sense if this function took in a string "key" of some sort.
And inside this function is a switch statement. The reasoning for this is It's possible a bad developer could use this function like `renderModal(true, false, true, false)` which is dumb, but could happen. You want to make it easier on a developer by removing as much "potential" problems with function usage.
With the switch statement, the usage becomes more straight forward: `renderModal("askDelete")`, `renderModal("confirmDelete")`, `renderModal("confirmSave")`, `renderModal("dismiss")` or something like that.

Reply:
And you can create string constants at the top of the file so it's clear what is available for use.
```
MODAL_CONFIRM_DELETE = "confirm_delete";
MODAL_DISMISS = "dismiss";

...
renderModal(MODAL_CONFIRM_DELETE);
```

Reply:
I'll put this in another PR.